ArduCopter/AP_Arming.cpp

```diff
@@ -233,52 +233,75 @@ bool AP_Arming_Copter::parameter_checks(bool display_failure)
+        }
+        if (!is_positive(copter.pos_control->get_accel_z_pid().kI())) {
             parameter_checks_pid_warning_message(display_failure, "PSC_ACCZ_I");
             return false;
-        } else if (is_zero(copter.attitude_control->get_rate_roll_pid().kP()) && is_zero(copter.attitude_control->get_rate_roll_pid().ff())) {
+        }
+        if (!is_positive(copter.attitude_control->get_rate_roll_pid().kP()) &&
+            !is_positive(copter.attitude_control->get_rate_roll_pid().ff())) {
             parameter_checks_pid_warning_message(display_failure, "ATC_RAT_RLL_P");
             return false;
-        } else if (is_zero(copter.attitude_control->get_rate_roll_pid().kI()) && is_zero(copter.attitude_control->get_rate_roll_pid().ff())) {
+        }
+        if (!is_positive(copter.attitude_control->get_rate_roll_pid().kI()) &&
+            !is_positive(copter.attitude_control->get_rate_roll_pid().ff())) {
             parameter_checks_pid_warning_message(display_failure, "ATC_RAT_RLL_I");
             return false;
-        } else if (is_zero(copter.attitude_control->get_rate_roll_pid().kD()) && is_zero(copter.attitude_control->get_rate_roll_pid().ff())) {
+        }
+        if (!is_positive(copter.attitude_control->get_rate_roll_pid().kD()) &&
+            !is_positive(copter.attitude_control->get_rate_roll_pid().ff())) {
             parameter_checks_pid_warning_message(display_failure, "ATC_RAT_RLL_D");
             return false;
-        } else if (is_zero(copter.attitude_control->get_rate_pitch_pid().kP()) && is_zero(copter.attitude_control->get_rate_pitch_pid().ff())) {
+        }
+        if (!is_positive(copter.attitude_control->get_rate_pitch_pid().kP()) &&
+            !is_positive(copter.attitude_control->get_rate_pitch_pid().ff())) {
             parameter_checks_pid_warning_message(display_failure, "ATC_RAT_PIT_P");
             return false;
-        } else if (is_zero(copter.attitude_control->get_rate_pitch_pid().kI()) && is_zero(copter.attitude_control->get_rate_pitch_pid().ff())) {
+        }
+        if (!is_positive(copter.attitude_control->get_rate_pitch_pid().kI()) &&
+            !is_positive(copter.attitude_control->get_rate_pitch_pid().ff())) {
             parameter_checks_pid_warning_message(display_failure, "ATC_RAT_PIT_I");
             return false;
-        } else if (is_zero(copter.attitude_control->get_rate_pitch_pid().kD()) && is_zero(copter.attitude_control->get_rate_pitch_pid().ff())) {
+        }
+        if (!is_positive(copter.attitude_control->get_rate_pitch_pid().kD()) &&
+            !is_positive(copter.attitude_control->get_rate_pitch_pid().ff())) {
             parameter_checks_pid_warning_message(display_failure, "ATC_RAT_PIT_D");
             return false;
-        } else if (is_zero(copter.attitude_control->get_rate_yaw_pid().kP()) && is_zero(copter.attitude_control->get_rate_yaw_pid().ff())) {
+        }
+        if (!is_positive(copter.attitude_control->get_rate_yaw_pid().kP()) &&
```

Comment:
This is certainly no worse than what we had but it doesn't actually protect against a single negative value.